src/com/ms/silverking/cloud/dht/collection/HybridPartialKeyCuckoo.java

```diff
@@ -2,96 +2,92 @@
+  }
+
+  class SubTable extends PKIntSubTableBase {
+    private final long[] ht;
+    private final int htOffset;
+
+    SubTable(long[] ht, int htOffset, int numBuckets, int entriesPerBucket, int keyShift) {
+      super(numBuckets, entriesPerBucket, keyShift);
+      this.ht = ht;
+      this.htOffset = htOffset;
     }
-    
-    public long[] getHashTableArray() {
-        return ht;
+
+    void clear() {
+      for (int i = 0; i < bufferSizeLongs; i++) {
+        //System.out.printf("%d\t%x\n", i, emptyEntry);
+        ht[htOffset + i] = emptyEntry;
+        //setBuf(i, emptyEntry);
+      }
     }
-    
-    class SubTable extends PKIntSubTableBase {
-        private final long[]    ht;
-        private final int       htOffset;
-        
-        SubTable(long[] ht, int htOffset, int numBuckets, int entriesPerBucket, int keyShift) {
-            super(numBuckets, entriesPerBucket, keyShift);
-            this.ht = ht;
-            this.htOffset = htOffset;
-        }
-        
-        void clear() {
-            for (int i = 0; i < bufferSizeLongs; i++) {
-                //System.out.printf("%d\t%x\n", i, emptyEntry);
-                ht[htOffset + i] = emptyEntry;
-                //setBuf(i, emptyEntry);
-            }
-        }
-
-        @Override
-        protected void setHT(int index, long offset) {
-            ht[htOffset + index] = offset;
-        }
-
-        @Override
-        protected long getHT(int index) {
-            //System.out.printf("%d\t%x\n", index, htBuf.get(index));
-            return ht[htOffset + index];
-        }
+
+    @Override
+    protected void setHT(int index, long offset) {
+      ht[htOffset + index] = offset;
     }
-    
-
-    private static HybridPartialKeyCuckoo rehash(PartialKeyIntCuckooBase oldTable) {
-        HybridPartialKeyCuckoo  newTable;
-        
-        newTable = new HybridPartialKeyCuckoo(oldTable.getConfig().doubleEntries(),
-                                              oldTable.getExternalStore());
-        try {
-            for (DHTKeyIntEntry entry : oldTable) {
-                //System.out.println(entry);
-                newTable.put(entry.getKey(), entry.getValue());
-            }
-            return newTable;
-        } catch (TableFullException tfe) {
-            throw new RuntimeException("Unexpected table full during rehash");
-        }
+
+    @Override
+    protected long getHT(int index) {
+      //System.out.printf("%d\t%x\n", index, htBuf.get(index));
+      return ht[htOffset + index];
+    }
+  }
+
+  private static HybridPartialKeyCuckoo rehash(PartialKeyIntCuckooBase oldTable) {
+    HybridPartialKeyCuckoo newTable;
+
+    newTable = new HybridPartialKeyCuckoo(oldTable.getConfig().doubleEntries(), oldTable.getExternalStore());
+    try {
+      for (DHTKeyIntEntry entry : oldTable) {
+        //System.out.println(entry);
+        newTable.put(entry.getKey(), entry.getValue());
+      }
+      return newTable;
+    } catch (TableFullException tfe) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)